### PR TITLE
Kokoro job configuration improvements

### PR DESCRIPTION
**Pull request by jtattermusch**
-- delete unused linux/grpc_master.cfg job
-- switch mac jobs to use RUN_TESTS_FLAGS env var (same as linux jobs)
-- switch windows jobs to use RUN_TESTS_FLAGS env var (same as linux jobs)
-- prepare for renaming grpc_master  suite to a more descriptive grpc_basictests (on windows and mac, similar change was done on linux recently).

### CODE REVIEW

**Comment by jtattermusch** [comment]
FYI @adelez

File: tools/internal_ci/windows/grpc_basictests.cfg [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Copyright 2017 gRPC authors.
```
Comment (matt-kwong) [inline tools/internal_ci/windows/grpc_basictests.cfg]: This whole file is just a copy of `grpc_master.cfg`. Delete it

File: tools/internal_ci/macos/grpc_basictests.cfg [hunk 2]
```diff
@@ -15,7 +15,7 @@
 # Config file for the internal CI (in protobuf text format)
```
Comment (matt-kwong) [inline tools/internal_ci/macos/grpc_basictests.cfg]: This whole file is just a copy of `grpc_master.cfg`. Is this supposed to be the PR version instead?

File: tools/internal_ci/macos/grpc_master.cfg [hunk 3]
```diff
@@ -15,11 +15,16 @@
...
 action {
   define_artifacts {
     regex: "**/*sponge_log.xml"
     regex: "github/grpc/reports/**"
   }
 }
+
+env_vars {
+  key: "RUN_TESTS_FLAGS"
+  value: "-f basictests macos --internal_ci -j 2 --inner_jobs 4"
```
Comment (adelez) [inline tools/internal_ci/macos/grpc_master.cfg]: Is it ok to set bq_result_table now?

File: tools/internal_ci/windows/grpc_basictests.cfg [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+action {
+  define_artifacts {
+    regex: "**/*sponge_log.xml"
+    regex: "github/grpc/reports/**"
+  }
+}
+
+env_vars {
+  key: "RUN_TESTS_FLAGS"
+  value: "-f basictests windows -j 1 --inner_jobs 8 --internal_ci"
```
Comment (adelez) [inline tools/internal_ci/windows/grpc_basictests.cfg]: Same here.

File: tools/internal_ci/macos/grpc_basictests.cfg (hunk 2 above)
Comment (matt-kwong) [inline tools/internal_ci/macos/grpc_basictests.cfg]: Just saw you deleted `grpc_master.cfg` for linux in favor for `grpc_basictests.cfg`. `grpc_master.cfg` can be deleted in a future PR, after updated in Kokoro.

File: tools/internal_ci/windows/grpc_basictests.cfg (hunk 1 above)
Comment (matt-kwong) [inline tools/internal_ci/windows/grpc_basictests.cfg]: Just saw you deleted `grpc_master.cfg` for linux in favor for `grpc_basictests.cfg`. `grpc_master.cfg` can be deleted in a future PR, after updated in Kokoro.

File: tools/internal_ci/macos/grpc_master.cfg (hunk 3 above)
Comment (matt-kwong) [inline tools/internal_ci/macos/grpc_master.cfg]: The Macs aren't configured to have access to our BQ results, but the Windows might have access already.

File: tools/internal_ci/windows/grpc_basictests.cfg (hunk 1 above)
Comment (jtattermusch) [inline tools/internal_ci/windows/grpc_basictests.cfg]: Yes, that is correct. Sorry for not making the explanation clearer.

File: tools/internal_ci/macos/grpc_master.cfg (hunk 3 above)
Comment (jtattermusch) [inline tools/internal_ci/macos/grpc_master.cfg]: Windows workers should have access to bigquery already, but I'm not sure if the google api client python package is installed there.
Also, we'd need to make sure to differentiate between master and pull requests jobs in order to use bq_result_table.
I'll look into that in a followup PR.

**Comment by jtattermusch** [comment]
Known failures:
https://github.com/grpc/grpc/issues/11188

**Pull request merged**